types/node.lua

```diff
@@ -271,50 +271,88 @@ function nodetypes.nnode:outputYourself (typesetter, line)
    end
 end
 
-function nodetypes.nnode:toText ()
+--- Returns the text content of the nnode.
+-- Contrary to the parent class, this is the actual text content of the node,
+-- not a text representation of the node.
+--
+-- @treturn string The text content of the nnode.
+function nnode:toText ()
    return self.text
 end
 
-nodetypes.unshaped = pl.class(nodetypes.nnode)
-nodetypes.unshaped.type = "unshaped"
+--- An unshaped node is a text node that has not been shaped yet.
+--
+-- Derived from `nnode`.
+--
+-- Properties is_unshaped is true.
+-- Note that is_nnode is NOT true, as an unshaped node is not a representable node yet.
+--
+-- @type unshaped
+
+local unshaped = pl.class(nnode)
+unshaped.type = "unshaped"
 
-function nodetypes.unshaped:_init (spec)
+--- Constructor
+--
+-- @tparam table spec A table with the properties of the unshaped node.
+
+function unshaped:_init (spec)
    self:super(spec)
    self.width = nil
 end
 
-function nodetypes.unshaped:__tostring ()
+function unshaped:__tostring ()
    return "U(" .. self:toText() .. ")"
 end
 
-getmetatable(nodetypes.unshaped).__index = function (_, _)
+getmetatable(unshaped).__index = function (_, _)
```

Comment:
What is this?
The commented code is not helpful.

Reply:
No idea. Would be tempted to have it removed, and see where SILE breaks, if it does.

Reply:
It does. A vast majority of our regression suite outputs blank space if I remove this node type and adjust anywhere unshaped pushes happen to push an hbox node. Notably much of the BiDi segmenting falls apart, your italic correction code would be removed, etc. Somehow most of our nodes are starting out as unshaped.

Reply:
Wow ^^